test/node/byte_utils.test.ts

```diff
@@ -544,6 +594,102 @@ describe('ByteUtils', () => {
     });
   });
 
+  describe('randomBytes fallback case', () => {
+    describe('web', function () {
+      let bsonWithNoCrypto;
+      before(function () {
+        bsonWithNoCrypto = loadBSONWithGlobal({
+          crypto: null,
+          // if we don't add a copy of Math here then we cannot spy on it for the test
+          Math: {
+            pow: Math.pow,
+            floor: Math.floor,
+            random: Math.random
+          }
+        });
+      });
+
+      after(function () {
+        sinon.restore();
+        bsonWithNoCrypto = null;
+      });
+
+      it('should fall back to Math.random implementation for random bytes if crypto is not present', () => {
+        const randomSpy = sinon.spy(bsonWithNoCrypto.Math, 'random');
+        new bsonWithNoCrypto.BSON.UUID();
+        // 16 is the length of a UUID
+        expect(randomSpy).to.have.callCount(16);
```

Comment:
```suggestion
        expect(randomSpy.called).to.be.true;
```
optional - the current assertion doesn't feel quite right.  What we care about is that `randomSpy` is called - we don't care about how many times it is called.  If it were me, I'd only assert that it was called

Reply:
this would apply to the other tests in this file, if you choose to take the suggestion

Reply:
A UUID will never change it's length and I'd want to know if Math.random was called more times than necessary, it does expand the scope of the test but it could catch something if the function is changed down the road to accidentally be invoked 1000s of times just to be used for 16 bytes of random values

Reply:
Ah yes fair point, let me update the title

Reply:
updated!